src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/ProtobufOtlpExporterPersistentStorageTransmissionHandler.cs

```diff
@@ -50,11 +50,9 @@ protected override bool OnSubmitRequestFailure(byte[] request, int contentLength
     {
         if (RetryHelper.ShouldRetryRequest(response, OtlpRetry.InitialBackoffMilliseconds, out _))
         {
-            byte[]? data = null;
-
-            if (data != null)
+            if (request != null)
```

Comment:
Is it really possible to have a `null` request here? Maybe switch to `Debug.Assert(request != null)`.